src/containerapp/azext_containerapp/_utils.py

```diff
@@ -1714,3 +1726,161 @@ def format_location(location=None):
+        elif host_os == "Darwin":
+            compressed_download_file_name = f"{compressed_download_file_name}-macos.tgz"
+        else:
+            raise Exception(f"Unsupported host OS: {host_os}")
+
+        exec_path = os.path.join(bin_folder, exec_name)
+        if os.path.exists(exec_path):
+            return exec_path
+
+        # Attempt to install the pack CLI
+        url = f"https://github.com/buildpacks/pack/releases/download/{pack_cli_version}/{compressed_download_file_name}"
+        with urlopen(url) as req:
+            compressed_file = io.BytesIO(req.read())
+            if host_os == "Windows":
+                with zipfile.ZipFile(compressed_file) as zip_file:
+                    for file in zip_file.namelist():
+                        if file.endswith(exec_name):
+                            with open(exec_path, "wb") as f:
+                                f.write(zip_file.read(file))
+            else:
+                with tarfile.open(fileobj=compressed_file, mode="r:gz") as tar:
+                    for tar_info in tar:
+                        if tar_info.isfile() and tar_info.name.endswith(exec_name):
+                            with open(exec_path, "wb") as f:
+                                f.write(tar.extractfile(tar_info).read())
+
+        # Add executable permissions for the current user if they don't exist
+        if not os.access(exec_path, os.X_OK):
+            st = os.stat(exec_path)
+            os.chmod(exec_path, st.st_mode | stat.S_IXUSR)
+
+        return exec_path
+    except Exception as e:
+        # Swallow any exceptions thrown when attempting to install pack CLI
+        logger.warning(f"Failed to install pack CLI: {e}\n")
+
+    return None
+
+
+def patchable_check(repo_tag_split: str, oryx_builder_run_img_tags, inspect_result):
+    tag_prop = parse_oryx_mariner_tag(repo_tag_split)
+    result = {
+        "targetContainerAppName": inspect_result["targetContainerAppName"],
+        "targetContainerName": inspect_result["targetContainerName"],
+        "targetContainerAppEnvironmentName": inspect_result["targetContainerAppEnvironmentName"],
+        "targetResourceGroup": inspect_result["targetResourceGroup"],
+        "targetImageName": inspect_result["image_name"],
+        "oldRunImage": repo_tag_split,
+        "newRunImage": None,
+        "id": None,
+    }
+    if tag_prop is None:
+        result["reason"] = "Image not based from a Mariner tag in mcr.microsoft.com/oryx/dotnet."
+        return result
+    elif len(str(tag_prop["version"]).split(".")) == 2:
+        result["reason"] = "Image is using a run image version that doesn't contain a patch information."
+        return result
+    repo_tag_split = repo_tag_split.split("-")
+    if repo_tag_split[1] == "dotnet":
+        matching_version_info = oryx_builder_run_img_tags[repo_tag_split[2]][str(tag_prop["version"].major) + "." + str(tag_prop["version"].minor)][tag_prop["support"]][tag_prop["marinerVersion"]]
```

Comment:
It is suggested to add an example for this joining.

Reply:
Updated: added example to show the string arithmetic.